### PR TITLE
Add support for HTTPS connections for jolokia plugin.

### PR DESCRIPTION
**Pull request by ersatzhero**
See title.

### CODE REVIEW

File: plugins/inputs/jolokia/jolokia.go [hunk 1]
```diff
@@ -14,7 +14,16 @@ import (
...
 
-type Server struct {
+type JolokiaServer struct {
+	Name     string
+	Host     string
+	Username string
+	Password string
+	Port     string
+	Https    bool
+}
```
Comment (phemmer) [inline plugins/inputs/jolokia/jolokia.go]: My $0.02: I think instead of a half-dozen config params, the plugin should just accept a URL.

Comment (ersatzhero) [inline plugins/inputs/jolokia/jolokia.go]: I would agree with you, if there wouldn't be that proxy mode for this plugin.
But still, i will have a look at it.

Comment (phemmer) [inline plugins/inputs/jolokia/jolokia.go]: Not sure I follow what the proxy has to do with it. But the proxy should be a URL too. All the code does is concatenate all the config params together and then parse it as a url.

Comment (sparrc) [inline plugins/inputs/jolokia/jolokia.go]: +1 to @phemmer, it would be preferable to just have the jolokia plugin accept a url

**Comment by ersatzhero** [comment]
I have changed the configuration, like suggested, but this removes the [proxy mode](https://github.com/influxdata/telegraf/issues/473).
Having both options would result in even more configuration params.

**Comment by sparrc** [comment]
@ersatzhero can you explain exactly why it's necessary to remove the proxy mode for this change? I'm not quite sure I follow

**Comment by ersatzhero** [comment]
Having both modes, agent and proxy, in this plugin and also follow the suggestion to have a single parameter where you would specify a URL would lead to the example config down bellow where you either have the URL or the combination of host and port for each server, depending on which mode is activated.

```
 # Read JMX metrics through Jolokia
 [[inputs.jolokia]]

   [jolokia.proxy]
   host = "jolokia.proxy.my-domain.org"
   port = "8000"
   
   ## List of servers exposing jolokia read service
   [[inputs.jolokia.servers]]
     name = "MyServer"
     host = my-server
     port = 443
     url = "https://my-server:443/jolokia"
     username = "username"
     password = "password"
```

The URL would be used to for agent mode - host and port for proxy mode.
This would lead to a lof of confusion, i think.

My code now is just to show that this would be a bad idea to just have the URL and the proxy mode.
Instead the HTTPS parameter in the first try is still my favorite.

Or i am getting something completly mixed up.

**Comment by sparrc** [comment]
I still don't follow, why can't you just parse out those parameters from the url and use it together with the proxy? maybe make the proxy a real url as well? see `url.Parse` in the Go standard library.

**Comment by ersatzhero** [comment]
Changed code, adapted test.

Tests fail, when they run on my local (windows) machine.
Can I get more output from CircleCI? Can't find the info what is going wrong with my test.

**Comment by sparrc** [comment]
from the test output:

```
run 'go fmt ./...' to format your source code.
```

**Comment by sparrc** [comment]
closing for lack of fixing the unit tests, please feel free to resubmit a PR when you can address that

**Pull request closed without merging**